samples/crypto/tls/README.rst

```diff
@@ -0,0 +1,197 @@
+
+Certificates in general
+-----------------------
+
+The sample supports certifactes that are signed with either RSA or ECDSA.
+By default, the sample is configured to use ECDSA certificate, but it can be changed to use RSA certificate by setting:
+
+.. code-block:: console
+
+   TEST_RSA_CERTIFICATE_ENABLED=y
+
+
+Certificates when running in non-secure domain
+----------------------------------------------
+
+The sample can be compiled as a non-secure firmware image, which automatically configures it to use TF-M.
+When compiled for non-secure, the sample will store its TLS certificates in the TF-M Protected Storage.
+
+**Warning: Non-secure applications only support ECDSA certificates at the moment, which is enforced in the configuration files for non-secure builds.**
+
+
+Cipher suites
+=============
+
+Most TLS v1.2 cipher suites are supported, with a few exceptions.
+
+* RSA is not supported in non-secure applications.
+* AES256 is not supported on nRF9160 in non-secure applications.
+
+
+TAP adapter
+===========
+The sample uses the `Ethernet over RTT driver <https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/drivers/eth_rtt.html/>`_
+in nRF Connect SDK to transfer Ethernet frames to the connected PC. The PC must parse the incoming RTT data,
+and dispatch these packets to a running TAP network device. This functionality is included for Linux in the attached `eth_rtt_link` executable.
+
+
+openssl
+=======
+The openssl command line interface is used to perform the peer network operations when testing this sample. It can act as either client or server with simple commands in a terminal.
+Examples are given in the `Testing as server`_ and `Testing as client`_ sections below.
+
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf5340dk_nrf5340_cpuappns, nrf9160dk_nrf9160ns
+
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/crypto/tls`
+
+.. include:: /includes/build_and_run.txt
+
+.. _Testing:
+
+Testing as server
+=================
+
+Follow these steps to test the PSA TLS sample as `server`:
+
+1. Start a terminal emulator like PuTTY and connect to the used COM port with the following UART settings:
+
+   * Baud rate: 115.200
+   * 8 data bits
+   * 1 stop bit
+   * No parity
+   * HW flow control: None
+#. Compile and program the application.
+#. Observe the logs from the application using the terminal emulator.
+#. Start the ``eth_rtt_link`` executable as superuser with your development kit's segger-id and the following ipv4 address as parameters:
+
+   .. code-block:: console
+
+      sudo ./eth_rtt_link --snr 960010000 --ipv4 192.0.2.2
+
+#. Use ``openssl`` to perform the `client` connection and handshake operation.
+
+   .. code-block:: console
+
+      openssl s_client -connect 192.0.2.1:4243 -cipher ECDHE-ECDSA-AES128-SHA256 -CAfile certs/ecdsa/root_cert.pem
```

Comment:
The line above uses `192.0.2.2`, but here it's `192.0.2.1`. It looks strange, but is it how it works?

Reply:
This is correct.
192.0.2.2 is the address of the TAP interface, while 192.0.2.1 is the address of the PSA TLS server.
I think the TAP interface ip address can be anything as long as it's in the same subnet as PSA TLS server.